lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.ts

```diff
@@ -105,12 +106,21 @@ export class AttachFileCloudWidgetComponent extends UploadCloudWidgetComponent i
         return (!this.hasFile || this.multipleOption) && !this.field.readOnly;
     }
 
-    onRemoveAttachFile(file: File | RelatedContentRepresentation | Node) {
+    onRemoveFile(file: File | RelatedContentRepresentation | Node) {
+        if(!file['id']) {
+            return;
+        }
         this.removeFile(file);
         if (file['id'] === this.selectedNode?.id) {
             this.selectedNode = null;
-            this.contentModelFormFileHandler();
         }
+        if(file['id'] === this.lastSentToViewers?.id) {
+            this.contentModelFormFileHandler(this.field.value.length > 0 ? this.field.value[0] : null);
+        }
+    }
+
+    onRetrieveFileMetadata(file: File| RelatedContentRepresentation | Node) {
```

Comment:
Missing space between the "|" for types, normally it should trigger the linter warnings/errors